src/osal.h

```diff
@@ -114,6 +116,45 @@ enum os_eth_type {
 typedef uint32_t os_ipaddr_t;
 typedef uint16_t os_ipport_t;
 
+/**
+ * Load a binary blob.
+ *
+ * @param file_index       In:    Index of binary blobs
+ * @param object           Out:   Struct to load
+ * @param size             In:    Size of struct to load
+ * @return  0  if the operation succeeded.
+ *          -1 if an error occurred.
+ */
+int os_load_blob(
+   int                     file_index,
+   void                    *object,
+   size_t                  size
+);
+
+/**
+ * Save a binary blob.
+ *
+ * @param file_index       In: Index of binary blobs
+ * @param object           In: Struct to save
+ * @param size             In: Size of struct to save
+ * @return  0  if the operation succeeded.
+ *          -1 if an error occurred.
+ */
+int os_save_blob(
+   int                     file_index,
+   void                    *object,
+   size_t                  size
+);
+
+/**
+ * Clear a binary blob.
+ *
+ * @param file_index       In: Index of binary blobs
+ */
+void os_clear_blob(
```

Comment:
Maybe os_delete_blob()? Clear might indicate that a cleared blob can be read and a "cleared" blob is read back